### ISSUE TITLE
Remove batches for validation loss 

### ISSUE THREAD

**Issue by balthazarneveu** [balthazarneveu/molecule-retrieval-using-nlp#7]
Problem: we train a network , validation uses batches (+ dataloader shuffles which could be discarded) but this does not reflect properly the validation accuracy (contrastive "cross entropy" shall be computed over the whole validation set, just like we do in the evaluation script)

Fix: reuse the evaluation functions, probably move the embeddings by batches back to CPU to reconstruct the whole validation embeddings before computing the full contrastive loss.

- [x] log lrap to tensorboard
- [x] log lrap to wandb
- [x] pick best epoch based on accuracy rather than validation loss
